File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Angular/src/app/interfaces/response/registerresponse.ts
import { State } from "../state";

export interface RegisterResponse {
    errorEmailNotValid: boolean;
    errorEmailInUse: boolean;
    errorUsernameInUse: boolean;
    errorUsernameEmpty: boolean;
    errorPasswordEmpty: boolean;
    errorPhoneNumberEmpty: boolean;
    errorCouldNotCreateUser: boolean;
    errorCouldNotGetDevice: boolean;
    errorCouldNotCreateBearer: boolean;
    success: boolean;
    errorAuthentication: boolean;
    errorItemNotFound: boolean;
    state: State | null;
}

File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Shared/ResponseJsons/RegisterResponse.cs
namespace BeltmanSoftwareDesign.Shared.ResponseJsons
{
    public class RegisterResponse : Response
    {
        public bool ErrorEmailNotValid { get; set; }
        public bool ErrorEmailInUse { get; set; }
        public bool ErrorUsernameInUse { get; set; }
        public bool ErrorUsernameEmpty { get; set; }
        public bool ErrorPasswordEmpty { get; set; }
        public bool ErrorPhoneNumberEmpty { get; set; }
        public bool ErrorCouldNotCreateUser { get; set; }
        public bool ErrorCouldNotGetDevice { get; set; }
        public bool ErrorCouldNotCreateBearer { get; set; }
    }
}
